spring-xd-dirt/src/main/java/org/springframework/xd/dirt/cluster/ContainerFilter.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.dirt.cluster;
+
+import java.util.List;
+
+import org.springframework.xd.module.ModuleDescriptor;
+
+/**
+ * A filter interface that applies filter for the given containers and returns
+ * eligible containers for deployment.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public interface ContainerFilter {
```

Comment:
We are going to try to refactor this to rely on composition. The main requirement was to contribute the logic to ensure that no more than one Spark Streaming module (driver) is being deployed to an XD Container. The idea was to have a more general contribution model here (we find any bean of type ContainerFilter), and that approach is not yet available for ContainerMatcher (find any bean of type ContainerMatcher and add). Will explore.